lib/browser/navigation.js

```diff
@@ -70,6 +70,28 @@ function urlAsserter(url, query) {
   return new Asserter(getAndCheckUrl);
 }
 
+/**
+ * Wait for document to be in a certain state
+ * https://github.com/admc/wd#waiting-for-something
+ * https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState
+ *
+ * @param {object} self
+ * @param {string} state
+ * @param {number} [timeout] - Timeout for js condition to complete (optional)
+ */
+function waitForDocumentState(self, state, timeout) {
+  timeout = timeout || 8000;
+  var expr;
+  if (state === 'interactive') {
+    expr = "var state; (state = document.readyState) === 'interactive' || state === 'complete';";
```

Comment:
Is reading `document.readyState` twice really expensive? Or why is this assigning to a variable?

Reply:
no it isn't expensive.